KarhooUISDK/CommonUI/BottomSheet/KarhooBottomSheet.swift

```diff
@@ -0,0 +1,158 @@
+                // If the content view runs the risk of going off screen either because of the amount of content or because a keyboard needs to be opened,
+                // then wrap that view inside a ScrollView before injecting it into this bottom sheet
+                content()
+                
+                Spacer()
+                    .frame(height: getBottomPadding())
+                    .animation(.easeOut(duration: UIConstants.Duration.medium))
+            }
+            .background(KarhooUI.colors.white.getColor()) // TODO: add proper background after color definition tweaking
+            .clipShape(
+                RoundedRectangle(
+                    cornerRadius: viewModel.cornerRadius,
+                    style: .continuous
+                )
+            )
+            .frame(
+                maxHeight: UIScreen.main.bounds.height - getTopPadding(),
+                alignment: .bottom
+            )
+            .layoutPriority(Double(UILayoutPriority.defaultHigh.rawValue))
+            .onTapGesture {
+                // Do nothing. This was added to intercept touches so that they aren't picked up by the main VStack tap gesture recognizer
+                // .allowsHitTesting(false) does not work for SwiftUI views presented from UIKit views
+            }
+        })
+        .colorScheme(.light) // TODO: Delete this line after dark mode modifications
+        .background(
+            KarhooUI.colors.black.getColor()
+                .opacity(UIConstants.Alpha.overlay)
+        )
+        .edgesIgnoringSafeArea([.all])
+        .offset(x: 0, y: offset.height)
+        .opacity(2 - Double(offset.height / 200))
+        .gesture(
+            DragGesture()
+                .onChanged({ gesture in
+                    if gesture.translation.height > 0 {
+                        offset = gesture.translation
+                    }
+                })
+                .onEnded({ _ in
+                    if offset.height > 100 {
+                        dismiss()
+                    } else {
+                        offset = .zero
+                    }
+                })
+        )
+        .onTapGesture {
+            dismiss()
+        }
+    }
+    
+    init(viewModel: BottomSheetViewModel, @ViewBuilder content: @escaping () -> Content) {
+        self.viewModel = viewModel
+        self.content = content
+    }
+    
+    private func getBottomPadding() -> CGFloat {
+        let safeArea = UIApplication.shared.windows.first?.safeAreaInsets.bottom ?? 0
+        if keyboard.currentHeight > 0 {
+            return keyboard.currentHeight
+        }
+        
+        return safeArea
+    }
+    
+    private func getTopPadding() -> CGFloat {
+        UIApplication.shared.windows.first?.safeAreaInsets.top ?? 0
+    }
+    
+    private func dismiss() {
+        if let callback = viewModel.callback {
+            callback(ScreenResult.cancelled(byUser: true))
```

Comment:
I think you should call generic dismiss method in `ViewModel` and view model should perform it on it's own